src/pages/contests/index.js

```diff
@@ -1,57 +1,197 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { graphql } from "gatsby";
 
-import { contestsByState } from "../../utils/filter";
-
 import ContestList from "../../components/ContestList";
 import DefaultLayout from "../../templates/DefaultLayout";
 
 export default function Contests({ data }) {
-  // @todo: implement global state management instead of props drilling
+  const [filteredContests, setFilteredContest] = useState(null);
   const [contestStatusChanges, updateContestStatusChanges] = useState(0);
-
+  const [isLoading, setIsLoading] = useState(false);
+  const contests = data.contests.edges;
   const updateContestStatus = () => {
     updateContestStatusChanges(contestStatusChanges + 1);
   };
 
-  const contests = data.contests.edges;
+  const sortContests = (array, status) => {
+    let statusObject = {
+      upcomingContests: [],
+      activeContests: [],
+      sponsorReview: [],
+      judging: [],
+      awarding: [],
+      reporting: [],
+      completed: [],
+      other: [],
+    };
+
+    array.forEach((element) => {
+      const statusAndIdObj = status.filter(
+        (el) => el.contestId === element.node.contestid
+      );
+      if (statusAndIdObj === []) {
+        return null;
+      }
+      const data = {
+        ...element.node,
+        status: statusObject[0]?.status,
+      };
+      switch (statusAndIdObj[0]?.status) {
+        case "Pre-Contest":
+          statusObject.upcomingContests.push(data);
+          break;
+        case "Preview week":
+          statusObject.upcomingContests.push(data);
+          break;
```

Comment:
Can use fall-through.

```suggestion
        case "Pre-Contest":
        case "Preview week":
          statusObject.upcomingContests.push(data);
          break;
```

Reply:
Oh interesting I didn't know we can do it !